### ISSUE TITLE
On-prem installation guide for traditional Python environment missing

### ISSUE THREAD

**Issue by adgud** [getsentry/sentry-docs#1381]
As in title - I notice that the installation guide for a on-prem non-Docker installation has been removed in this PR: https://github.com/getsentry/sentry-docs/pull/1369

Can you explain why? Is it no longer supported? Should existing installations be migrated to Docker?

**Comment by BYK** [comment]
We have the following for anything else than Docker now:

>If you do not wish to use the Docker images we provide, you can still find [Sentry on PyPI](https://pypi.org/project/sentry/), that said this method is not recommended. You'll need to work your way back from [the main Sentry image](https://github.com/getsentry/sentry/blob/master/docker/Dockerfile) and [the service composition](https://github.com/getsentry/onpremise/blob/master/docker-compose.yml). It is not too hard but you are likely to spend a lot more time and hit some bumps ahead.

We are focusing on Docker-based deploys for repeatability and no matter what we put in the Python-based guide, it will get old pretty quickly. It will also not be supported actively as it is a hassle to setup and maintain that way multiple times to keep it active.

What are you specifically looking for @adgud?

**Comment by adgud** [comment]
I was looking for requirements/dependencies necessary for running Sentry without Docker. I found the information you quoted, but the main Sentry image definition is quite complex to be honest. Service composition expanded as well - for Sentry 9 there was (AFAIK) no need for Snuba, Clickhouse, Kafka, Zookeeper - will I'll be able to continue using Sentry on-prem without those?

I feel comfortable with running my existing Sentry installation which looks much simpler than the Docker stack of services.

My question still stands - should existing installations be migrated to Docker? If future upgrades may break my current installation then I'll switch to Docker-based one.

**Comment by BYK** [comment]
>but the main Sentry image definition is quite complex to be honest.

Would love to make it more understandable. Are there any pieces you find especially hard to follow so we can work on those to improve?

>Service composition expanded as well - for Sentry 9 there was (AFAIK) no need for Snuba, Clickhouse, Kafka, Zookeeper - will I'll be able to continue using Sentry on-prem without those?

Yes, this unfortunately is here to stay and you won't be able to run Sentry with these surrounding services now. That's the main reason for us pushing forward with the Docker-based approach as detailing these services and how they integrate in prose (in docs) and keeping this up to date is a lot harder and inefficient than to simply put them in a working docker-compose example that is continuously tested.

>I feel comfortable with running my existing Sentry installation which looks much simpler than the Docker stack of services.

That won't be the case if you want to run Sentry 10 yourself as I touched on above.

>My question still stands - should existing installations be migrated to Docker? If future upgrades may break my current installation then I'll switch to Docker-based one.

It is not us to say which way you should be going. All I can say is we won't be actively supporting the non-Docker installation path from v10 and onward and you'd probably have an easier time if you stick to the Docker-based flows.

Happy to make any improvements to the docs or the code to make things easier for you if you can point to specific areas that need improvements from your perspective.

**Comment by adgud** [comment]
If Sentry 10 requires all these surrounding services then I agree that Docker-based approach is much better :+1: 

>That won't be the case if you want to run Sentry 10 yourself as I touched on above.

Okay, how about staying on Sentry 9? Will 9 be maintained for a while or are users expected to upgrade to 10 as soon as it is released? By "maintained" I mean security updates and maybe bugfixes.

Regarding complexity of the Dockerfile - I will need to dig in a bit deeper into it before I can provide you with valuable feedback, but for this I will create a separate issue in the future (or maybe even a PR :smiley:  ).

**Comment by BYK** [comment]
>Okay, how about staying on Sentry 9? Will 9 be maintained for a while or are users expected to upgrade to 10 as soon as it is released? By "maintained" I mean security updates and maybe bugfixes.

We don't have any plans for releasing a new version on the 9.1 line that said if a serious security issue is discovered, we may consider it based on the usage. I'd assume v9 to be unmaintained for all practical purposes though.

>Regarding complexity of the Dockerfile - I will need to dig in a bit deeper into it before I can provide you with valuable feedback, but for this I will create a separate issue in the future (or maybe even a PR 😃 ).

That would be great, thanks a lot! We certainly want to make this accessible, not opaque and feedback like yours help us get there :)

Do you think we can close this issue now or are there other things you'd like to get addressed?

**Comment by adgud** [comment]
That is all, thanks for clearing this up for me - I'll close this issue.
